### PR TITLE
yusuf

### PR DESCRIPTION
**Pull request by yusufkaratoprak**
- update for Getting traffic in using TLS and LetsEncrypt certificates
- update cer manager side for edge stack
- How to use Ambassador to host multiple TLS-enabled hosts on the same cluster
- How to use Ambassador to host multiple TLS-enabled hosts on the same cluster-2
- make ambassador edge stak with details and simplified
- make ambassador edge stak with details and simplified
- make ambassador edge stak with details and simplified
- added more details about service,deploy etc. yml
- added more details about service,deploy etc. yml 2
- added more details about service,deploy etc. yml 2
- enrichment content for edge stack,mapping etc.
- enrichment content for edge stack,mapping etc.
- enrichment mapping content
- enrichment mapping content
- enrichment mapping content
- update with yml structure and detailed and drop unnecessary parts etc.
- update with yml structure and detailed and drop unnecessary parts etc.
- enabling proxy protocol updated
- added mapping  updated
- added curl request for testing
- update for meaningful service name reducing name entanglements
- update for meaningful service name reducing name entanglements
- update for meaningful service name reducing name entanglements
- update for meaningful service name reducing name entanglements
- enrich document by adding more clear sentence gramatically improvements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -171,630 +171,476 @@ curl https://raw.githubusercontent.com/grafana/helm-charts/main/charts/loki-stac
...
-X-Envoy-Expected-Rq-Timeout-Ms: 3000
-X-Envoy-Original-Path: /echo/
-Content-Length: 0
 
-```
+### Configure your domain mapping to point the domains to the cluster LB
 
-#### Getting traffic in using DNS name
-This is fairly straightforward. We will just add an A record for every host into the DNS. In my case, I added 2 hosts (echo and test) pointing to the same LB.
+Here  hosting 2 hosts (echo.mandrakee.xyz, quote.mandrakee.xyz) on the same cluster.
```
Comment (bikram20) [inline README.md:292]: You can add the A-record either using doctl, or from the UI.
https://docs.digitalocean.com/products/networking/dns/how-to/manage-records/

File: README.md [hunk 2]
```diff
@@ -171,630 +171,476 @@ curl https://raw.githubusercontent.com/grafana/helm-charts/main/charts/loki-stac
...
+158200732    SOA     @       1800                    0           0       1800    0  
+158200733    NS      @       ns1.digitalocean.com    0           0       1800    0  
+158200734    NS      @       ns2.digitalocean.com    0           0       1800    0  
+158200735    NS      @       ns3.digitalocean.com    0           0       1800    0  
+158200782    A       echo    143.244.208.164         0           0       180     0  
+158201299    A       quote   143.244.208.164         0           0       180     0  
+~   
+~ kgsvcn ambassador   
+NAME               TYPE           CLUSTER-IP       EXTERNAL-IP       PORT(S)                      AGE  
+ambassador         LoadBalancer   10.245.215.249   143.244.208.164   80:32723/TCP,443:31416/TCP   23h
```
Comment (bikram20) [inline README.md:306]: Hide the external IP in the output. Just mark it as xxx.xxx.

File: README.md [hunk 3]
```diff
@@ -171,630 +171,476 @@ curl https://raw.githubusercontent.com/grafana/helm-charts/main/charts/loki-stac
...
+  namespace: ambassador
+spec:
+  prefix: /echo/
+  host: echo.mandrakee.xyz
+  service: echo
+  ```
 
-tokenSecretRef:
+### Enabling Proxy Protocol
+There are lot of configuration options available to you when running Ambassador Edge Stack but we will focus on automatic certificate management in the Ambassador Edge Stack in this part. After deploying the Service above  and manually enabling the proxy protocol you have to need to deploy the following Ambassador Module to manage Ambassador Edge Stack to utilize the proxy protocol and then restart Ambassador Edge Stack for the configuration to take effect.
```
Comment (bikram20) [inline README.md:462]: I would remove the first sentence. That has nothing to do with proxy protocol.

Start with: "Proxy protocol enables a L4 Load balancer to communication the original client IP. For this work, we need to configure both ends (DO LB and AES).

File: README.md [hunk 4]
```diff
@@ -171,630 +171,476 @@ curl https://raw.githubusercontent.com/grafana/helm-charts/main/charts/loki-stac
...
 
-name: digitalocean-dns
+Also you can create load balancer by using configuration like below
```
Comment (bikram20) [inline README.md:464]: You already added an LB in the beginning. When you installed AES, a DO LB was automatically created.

What we're trying to do is to configure proxy protocol on that DO LB.
https://docs.digitalocean.com/products/kubernetes/how-to/configure-load-balancers/#proxy-protocol

Then we should configure AES Proxy protocol.
 config:
     use_proxy_proto: true

That's it.

**Review by bikram20 — COMMENTED** [review]
It looks good to me, minus some minor issues. Please review the proxy protocol section, and other comments I put.

File: README.md [hunk 5]
```diff
@@ -171,630 +171,476 @@ curl https://raw.githubusercontent.com/grafana/helm-charts/main/charts/loki-stac
...
-*  AmbassadorMapping resource maps a resource (group of URLs with the common prefix) to a backend service. At the most basic level, you just forward a prefix to a backend service. You can also configure auth, circuit breaking, automatic retries, header based routing, method based routing, regex on the header, request rewrite, traffic shadowing, rate limiting etc.
+More Details and extended explanation please visit: [The Ambassador Edge Stack Architecture](https://www.getambassador.io/docs/edge-stack/2.0/topics/concepts/architecture/) 
 
-Let us configure a most basic forwarding by using the above resources.
+The set of configuration steps are as follows
+- Install AES.
+- Configure 2 hosts (domain1, domain2) on the cluster. For this example, we're using quote.mandake.xyz, and echo.mandrake.xyz as 2 differents hosts on the same cluster.
+- Enable different paths for the domains. For quote.mandrake.xyz, we create a backend service called quote. For echo.mandrake.xyz, we create a echo service called echo.
+- Domains are configured with TLS, and have different URL paths.
+- Verify the installation.
```
Comment (bikram20) [inline README.md:218]: We should insert the network diagram (slide) as a picture.

File: README.md (hunk 2 above)
Comment (yusufkaratoprak) [reply README.md:306]: done!

File: README.md (hunk 3 above)
Comment (yusufkaratoprak) [reply README.md:462]: done!

File: README.md (hunk 5 above)
Comment (yusufkaratoprak) [reply README.md:218]: I added screenshot : https://docs.google.com/presentation/d/1ECcO4hcpwgHyNrkkKaGOkG-5SfZWOi2jyRv1j8j2MkM/edit#slide=id.p

File: README.md (hunk 1 above)
Comment (yusufkaratoprak) [reply README.md]: done

**Comment by yusufkaratoprak** [comment]
updated with improvements from PR review.

File: README.md (hunk 4 above)
Comment (yusufkaratoprak) [reply README.md]: done

**Pull request closed without merging**